tensorflow_model_optimization/python/core/sparsity_tf2/riglpruner_test.py

```diff
@@ -0,0 +1,577 @@
+        dtype=weight_dtype)
+    grad = self.grad(weight)
+
+    p = pruner.RiGLPruner(
+      update_schedule=self.skip_updater,
+      sparsity=self.target_sparsity,
+      block_size=self.block_size,
+      block_pooling_type=self.block_pooling_type,
+      seed=self.seed,
+      noise_std=self.noise_std,
+      reinit=self.reinit
+    )
+
+    optimizer = tf.keras.optimizers.SGD(learning_rate=0.01)
+    optimizer.iterations.assign(0)
+    p.create_slots(optimizer, weight)
+
+    n_prune = 25 # drop ratio 0.5 of 50% sparse
+    n_total = 100
+    n_keep = n_total - n_prune
+    current_mask = optimizer.get_slot(weight, 'mask').read_value()
+    dropped_mask = p._generic_top_k(weight * current_mask, current_mask, n_keep,  n_total)
+    grow_scores = tf.math.abs(grad)
+    grow_scores = tf.reshape(grow_scores, (-1,))
+    # set enabled connections (ones) to min(scores) - 1, i.e. they have the lowest scores
+    grow_scores_lifted = tf.where(
+      tf.math.equal(dropped_mask, 1),
+      tf.ones_like(dropped_mask) * (tf.reduce_min(grow_scores) - 1), grow_scores
+    )
+    grown_mask = p._generic_top_k(grow_scores_lifted, current_mask, n_prune,  n_total)
+    self.assertAllEqual(tf.reduce_sum(dropped_mask * grown_mask), 0.)
+
+
+  def testMaskNoChangeBeforeandAfter(self):
+    weight = tf.Variable(np.linspace(1.0, 100.0, 100))
+    weight_dtype = weight.dtype.base_dtype
+    mask = tf.Variable(
+        tf.ones(weight.get_shape(), dtype=weight_dtype),
+        dtype=weight_dtype)
+    sparse_vars = [(mask, weight, self.grad(weight))]
+
+    p = pruner.RiGLPruner(
+      update_schedule=self.skip_updater,
+      sparsity=self.target_sparsity,
+      block_size=self.block_size,
+      block_pooling_type=self.block_pooling_type,
+      seed=self.seed,
+      noise_std=self.noise_std,
+      reinit=self.reinit
+    )
+
+    optimizer = tf.keras.optimizers.SGD(learning_rate=0.01)
+    optimizer.iterations.assign(0)
+    p.create_slots(optimizer, weight)
+
+    mask_before_update = optimizer.get_slot(weight, 'mask').read_value()
+    before_sparsity = np.count_nonzero(mask_before_update) / tf.size(mask_before_update)
+    self.assertAllEqual(before_sparsity, self.target_sparsity)
+
+    next_step = optimizer.iterations.assign_add(1)
+    reset_momentum, new_connections = p.update_masks(sparse_vars, next_step)
+    self.assertAllEqual(reset_momentum, False)
+    expected_new_connections = tf.zeros((), dtype=tf.bool)
+    self.assertAllEqual(new_connections, expected_new_connections)
+
+    mask_after_update = optimizer.get_slot(weight, 'mask').read_value()
+    after_sparsity = np.count_nonzero(mask_after_update) / tf.size(mask_before_update)
+    self.assertAllEqual(after_sparsity, self.target_sparsity)
+    self.assertAllEqual(mask_after_update, mask_before_update) # no update
+
+  @parameterized.parameters((3, 7, 2), (1, 5, 3), (0, 4, 1))
+  def testDropFractionZeroNoMaskChange(self, begin_step, end_step, freq_update):
```

Comment:
Can you add a comment explaining this test? As far as I can tell it's setting an argument in the schedule saying stuff should never be dropped? So we expect the masks will never be updated?